test/slax_web/websockets/issue_test.exs

```diff
@@ -0,0 +1,53 @@
+defmodule SlaxWeb.Issue.Test do
+  use SlaxWeb.ConnCase, async: true
+  import Mox
+
+  alias SlaxWeb.Issue
+
+  setup :verify_on_exit!
+
+  test "ignores non relevant events", _ do
+    event1 = %{"subtype" => "bot_message"}
+    event2 = %{"subtype" => "message_changed"}
+    event3 = %{"type" => "not_a_message"}
+    event4 = %{"bot_id" => "beep123"}
+
+    assert Issue.handle_event(event1) == nil
+    assert Issue.handle_event(event2) == nil
+    assert Issue.handle_event(event3) == nil
+    assert Issue.handle_event(event4) == nil
+  end
+
+  test "test regex", _ do
+    assert [["slax-test#1", "", "slax-test", "#1"]] ==
+             Issue.scan_text("some extra text slax-test#1 some extra text")
+
+    assert [["slax-test#1", "", "slax-test", "#1"], ["test#2", "", "test", "#2"]] ==
+             Issue.scan_text("slax-test#1 some extra text test#2")
+
+    assert [["org/repo#1", "org/", "repo", "#1"]] == Issue.scan_text("garbage org/repo#1 garb")
+
+    assert [] == Issue.scan_text("test#number")
+    assert [] == Issue.scan_text("test #number")
+    assert [] == Issue.scan_text("test # number")
+    assert [] == Issue.scan_text("#number")
+    assert [] == Issue.scan_text("#1")
+    assert [] == Issue.scan_text("test #1")
+    assert [] == Issue.scan_text("/1")
+    assert [] == Issue.scan_text("test/1")
+  end
+
+  #  TODO: For when Tentacat is mocked
+  # test "send issue", %{conn: conn} do
+  #   event1 = %{"thread_ts" => 123, "channel" => 123, "text" => "test#1", "type" => "message"}
+  #   event2 = %{"channel" => 123, "text" => "test#1", "type" => "message"}
+
+  #   Slax.HttpMock
+  #   |> expect(:post, 2, fn _, _, _, _ ->
+  #     {:ok, %HTTPoison.Response{status_code: 200, body: ~s<{"ok": true}>}}
+  #   end)
+
+  #   Issue.handle_event(event1)
+  #   Issue.handle_event(event2)
+  # end
```

Comment:
Two random comments about this:
- I tend to be really ruthless about what I leave in the codebase in terms of commented-out code. If I'm keeping it b/c I might need it later, I like to be really really sure that (a) I will definitely need it very soon and (b) it would be nontrivial to write from scratch when that time comes. For me, this commented code wouldn't quite get there.
- When leaving a TODO in the codebase, I like to make sure it has a link to the Github issue where it will be addressed. This makes sure we _are_ tracking the work outside the code comment (b/c code comments are a terrible way to track TODOs) and also serves as a way to _find_ the code when I need to work that issue (search the codebase for the issue link.) But also, similar to the above, I tend to be pretty ruthless about whether I need a TODO in the code at all. If the issue is there, and this spot in the code will be self-evident to whoever picks up that issue, then we probably don't need it.

Reply:
Thanks for the comments, I like the linking TODOs to the issue number and mentioning it in the issue I hadn't thought about that before. In this case yeah I agree the code is trivial enough to just rewrite when the issue is being worked